client/pages/sections/protocols/protocols.js

```diff
@@ -0,0 +1,128 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import classnames from 'classnames';
+
+import size from 'lodash/size';
+import ProtocolSections from './protocol-sections';
+
+import Fieldset from '../../../components/fieldset';
+import Repeater from '../../../components/repeater';
+import Controls from '../../../components/controls';
+
+class Form extends PureComponent {
+  removeItem = e => {
+    e.preventDefault();
+    this.props.removeItem();
+  }
+
+  render() {
+    const { index, name, updateItem, exit, toggleActive, prefix = '', ...props } = this.props;
+    return (
+      <div className={classnames('protocol', 'panel')}>
+        {
+          index !== 0 && (
+            <a href="#" className="float-right" onClick={this.removeItem}>Remove</a>
+          )
+        }
+
+        <h2>{`Protocol ${index + 1}`}</h2>
+        <Fieldset
+          { ...props }
+          fields={props.fields}
+          prefix={`${prefix}${name}-${index}-`}
+          onFieldChange={(key, value) => updateItem({ [key]: value })}
+        />
+        <Controls onContinue={toggleActive} onExit={exit} exitDisabled={true} />
+      </div>
+    );
+  }
+}
+
+class Protocol extends PureComponent {
+  state = {
+    active: !this.props.values.title,
+    complete: false
+  }
+
+  toggleActive = () => {
+    this.setState({ active: !this.state.active });
+  }
+
+  getParts = () => {
+    if (!this.props.editable || !this.props.location.hash) {
+      return null;
+    }
+    const parts = this.props.location.hash.split('.');
+    if (!parts.length) {
+      return null;
+    }
+    return parts;
+  }
+
+  isActive = parts => {
+    if (!parts) {
+      return false;
+    }
+    const active = parseInt(parts[1], 10);
+    return isNaN(active) ? false : active === this.props.index;
+  }
+
+  render() {
+    const { editable } = this.props;
+
+    const parts = this.getParts();
+    const isActive = this.isActive(parts);
+
+    return editable && this.state.active
+      ? <Form
+        {...this.props}
+        toggleActive={this.toggleActive}
+      />
+      : <ProtocolSections
+          {...this.props}
+          parts={isActive && parts}
+          onToggleActive={this.toggleActive}
+        />
+  }
+}
+
+class Protocols extends PureComponent {
+  save = protocols => {
+    this.props.save({ protocols });
+  }
+
+  edit = e => {
+    e.preventDefault();
+    this.props.retreat();
+  }
+
+  render() {
+    const { project, editable } = this.props;
```

Comment:
It doesn't look like we need anything other than `project.protocols` here to actually render, so is there a performance benefit to only returning the protocols from `mapStateToProps`?

Reply:
I've tried to trace through the downstream components that get rendered with `{...props}` and none of them seem to obviously be using the project.

Reply:
right you are